# Nextflow RNA-Seq Analysis Pipeline

This Nextflow script automates the process of downloading an Ensembl GTF file, generating feature counts from a list of BAM files, and performing DESeq2 analysis in R for RNA-Seq data analysis.

## Prerequisites

Before using this script, ensure that you have the following prerequisites installed:

- [Nextflow](https://www.nextflow.io/docs/latest/getstarted.html)
- [R](https://www.r-project.org/)
- [DESeq2 R package](https://bioconductor.org/packages/release/bioc/html/DESeq2.html)
- [featureCounts](http://bioinf.wehi.edu.au/featureCounts/)

## Usage

1. Clone this repository or download the `nextflow_script.nf` file.

2. Run the Nextflow script with the desired parameters. You can specify the following parameters:

   - `ensemblRelease`: Ensembl release version (default: 'release-93').
   - `bamFilesDir`: Directory containing BAM files (default: current directory).
   - `outputDir`: Directory for storing output files (default: current directory).
   - `coldata`: Path to coldata.csv (default: './coldata.csv').
   - `designFormula`: Design formula for DESeq2 analysis (default: '~ group').

   Example usage:

   ```
   nextflow run nextflow_script.nf --ensemblRelease release-94 --bamFilesDir /path/to/bam/files --outputDir /path/to/output --coldata /path/to/SampleInformation.csv --designFormula '~ condition'
    ```

The script will automatically download the Ensembl GTF file, run featureCounts on the specified BAM files, and perform DESeq2 analysis. The normalized counts will be stored in the specified outputDir.

## Output

The script generates the following output files:

- counts.txt: Feature counts generated by featureCounts.
- norm_counts.csv: Normalized counts from DESeq2 analysis.
